MultiSlider.js

```diff
@@ -1,573 +1,486 @@
-    if (!this.props.enabledOne) {
+  let moveOne = gestureState => {
+    if (enabledOne) {
       return;
     }
 
-    const accumDistance = this.props.vertical
-      ? -gestureState.dy
-      : gestureState.dx;
-    const accumDistanceDisplacement = this.props.vertical
+    let accumulatedDistance = vertical ? -gestureState.dy : gestureState.dx;
+    let accumulatedDistanceDisplacement = vertical
       ? gestureState.dx
       : gestureState.dy;
-
-    const unconfined = I18nManager.isRTL
-      ? this.state.pastOne - accumDistance
-      : accumDistance + this.state.pastOne;
-    var bottom = 0;
-    var trueTop =
-      this.state.positionTwo -
-      (this.props.allowOverlap
+    let unconfined = I18nManager.isRTL
+      ? pastOne - accumulatedDistance
+      : accumulatedDistance + pastOne;
+    let bottom = 0;
+    let trueTop =
+      positionTwo -
+      (allowOverlap
         ? 0
-        : this.props.minMarkerOverlapDistance > 0
-        ? this.props.minMarkerOverlapDistance
-        : this.stepLength);
-    var top = trueTop === 0 ? 0 : trueTop || this.props.sliderLength;
-    var confined =
+        : minMarkerOverlapDistance > 0
+        ? minMarkerOverlapDistance
+        : stepLength);
+    let top = trueTop === 0 ? 0 : trueTop || sliderLength;
+    let confined =
       unconfined < bottom ? bottom : unconfined > top ? top : unconfined;
-    var slipDisplacement = this.props.touchDimensions.slipDisplacement;
+    let slipDisplacement = touchDimensions.slipDisplacement;
 
     if (
-      Math.abs(accumDistanceDisplacement) < slipDisplacement ||
+      Math.abs(accumulatedDistanceDisplacement) < slipDisplacement ||
       !slipDisplacement
     ) {
-      var value = positionToValue(
-        confined,
-        this.optionsArray,
-        this.props.sliderLength,
-      );
-      var snapped = valueToPosition(
-        value,
-        this.optionsArray,
-        this.props.sliderLength,
-      );
-      this.setState({
-        positionOne: this.props.snapped ? snapped : confined,
-      });
-
-      if (value !== this.state.valueOne) {
+      let value = positionToValue(confined, optionsArray, sliderLength);
+      let snapped = valueToPosition(value, optionsArray, sliderLength);
+
+      setPositionOne(snapped ? snapped : confined);
+
+      if (value !== valueOne) {
         this.setState(
           {
             valueOne: value,
           },
           () => {
-            var change = [this.state.valueOne];
-            if (this.state.valueTwo) {
-              change.push(this.state.valueTwo);
+            let change = [valueOne];
+            if (valueTwo) {
+              change.push(valueTwo);
             }
-            this.props.onValuesChange(change);
+            onValuesChange(change);
 
-            this.props.onMarkersPosition([
-              this.state.positionOne,
-              this.state.positionTwo,
-            ]);
+            onMarkersPosition([positionOne, positionTwo]);
           },
```

Comment:
```jsx
setValueOne(value);
let change = [value];
if (valueTwo) {
   change.push(valueTwo)
}
onValuesChange(change);
onMarkersPosition([positionOne, positionTwo]);
```

Reply:
First, I have to extract somehow setting the state from the `if` statement or to break somehow the `moveOne` function.